lib/xenops_utils.ml

```diff
@@ -617,3 +617,40 @@ let chunks size lst =
 let best_effort txt f =
   try f ()
   with e -> info "%s: ignoring exception %s" txt (Printexc.to_string e)
+
+
+let json_length rpc = rpc |> Jsonrpc.to_string |> String.length
+let xml_length rpc =
+  let xml = rpc |> Xmlrpc.to_string in
+  let length = ref (String.length xml) in
+  (* protect_fn in xml_spaces in XAPI escapes some additional characters *)
+  String.iter (function
+        | ' ' | '\t' | '\n' | '\r' | '%' -> incr length
+        | _ -> ()
+    ) xml;
+  !length
+
+(* It is stored as Json in internal communication, but as xml in the database *)
+let char_max_encoded_length =
+  Array.init 256 (fun i ->
+      if i >= 0x80 then 3 (* Uutf.u_rep, see utf8_recode *)
+      else
+      let s = String.init 1 (fun _ -> Char.chr i) in
+      let rpc = Rpc.String s in
+      let empty = Rpc.String "" in
+      max
+        (json_length rpc - json_length empty)
+        (xml_length rpc - xml_length empty)
+  )
+
+let str_max_encoded_length str =
+  let s = ref 0 in
+  String.iter (fun c -> s := !s + char_max_encoded_length.(Char.code c)) str;
+  !s
+
+let longest_encoded_char = Array.fold_left max 0 char_max_encoded_length
+
+let entry_overhead = String.length "(''%.'')%."
+
+let entry_size_bytes key value =
```

Comment:
Please add a comment explaining the context and perhaps an example.

Reply:
The background is finding the size of the encoding in XML but a small  block comment at the beginning might help.

Reply:
Added an explanation